### PR TITLE
Laravel home v1.3

### PR DESCRIPTION
**Pull request by SterbenMy**

### CODE REVIEW

File: app/Http/Controllers/ArticleController.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+namespace App\Http\Controllers;
+
+use App\Models\Article;
+use App\Models\ArticleTag;
+
+
+class ArticleController extends Controller
+{
+    public function index($id){
+//        $tag=DB::table('tags')
```
Comment (ghost) [inline app/Http/Controllers/ArticleController.php:12]: you should not keep commented code in you project, please remove it

File: app/Http/Controllers/ArticleController.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace App\Http\Controllers;
+
+use App\Models\Article;
+use App\Models\ArticleTag;
+
+
+class ArticleController extends Controller
+{
+    public function index($id){
```
Comment (ghost) [inline app/Http/Controllers/ArticleController.php:11]: usually index name is used for search, not for show and article

File: app/Http/Controllers/ArticleController.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+//            ->select('tags.name','tagsConnections.tag_id','tagsConnections.article_id')
+//            ->join('tagsConnections','tags.id',"=",'tagsConnections.tag_id');
+//
+//        $tags=Article::select('articles.id','articles.title','tg.tag_id','tg.name as tagName')
+//            ->joinSub($tag,'tg',function ($join){
+//            $join->on('tg.article_id',"=",'articles.id');
+//        })->where('articles.id',$id);
+
+
+        $tg=
```
Comment (ghost) [inline app/Http/Controllers/ArticleController.php:22]: What does `tg` mean? Please use descriptive variable names

File: app/Http/Controllers/ArticleController.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+            $tags=Article::select('articles.id',
+                                  'articles.title',
+                                  'tg.tag_id',
+                                   'tg.name as tagName')
+                ->joinSub($tg,'tg',function($join){
+                    $join->on('tg.article_id',"=",'articles.id');
+        })->where('articles.id',"=",$id)->get();
+
+        $article = Article::select('articles.id',
```
Comment (ghost) [inline app/Http/Controllers/ArticleController.php:37]: extract each select in separated method

File: app/Http/Controllers/ContactFormController.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        $this->mailer = $mailer;
+    }
+
+    public function index()
+    {
+        return view('mail.email');
+    }
+
+    public function sendMailas(Request $request)
```
Comment (ghost) [inline app/Http/Controllers/ContactFormController.php:28]: sendMail

File: app/Http/Controllers/ContactFormController.php [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    {
+
+        $request->validate([
```
Comment (ghost) [inline app/Http/Controllers/ContactFormController.php:31]: extract validation in separated request class

File: app/Http/Controllers/ContactFormController.php [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            function(Message $message) use ($data) {
+            $message->to($data['email'])
+                    ->subject($data['subject'])
+                    ->from('cristi@cristi.com',$data['name']);
+        });
+
+        return back()->with(['message' => 'Email successfully sent!']);
+    }
+
+//    public function __invoke()
```
Comment (ghost) [inline app/Http/Controllers/ContactFormController.php:56]: remove this code

File: app/Services/FirstRequestLogger.php [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+
+namespace App\Services;
+
```
Comment (ghost) [inline app/Services/FirstRequestLogger.php:5]: to many empty lines, one is enough

File: database/factories/ArticleFactory.php [hunk 9]
```diff
@@ -22,18 +22,18 @@ class ArticleFactory extends Factory
     public function definition()
     {
         return [
-            'title' => $this->faker->name,
+            'title' => $this->faker->title,
```
Comment (ghost) [inline database/factories/ArticleFactory.php:25]: title has `Mr`, `Miss` and so on, I propose to use sentence

File: resources/views/footer.blade.php [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+                    <li><a href="https://www.linkedin.com"><i class="fab fa-linkedin-in"></i></a></li>
+                    <li><a href="https://www.instagram.com"><i class="fab fa-instagram"></i></a></li>
+                </ul>
+            </div>
+
+            <div class="col-md-6 link_footer">
+                <h4>Quick Links</h4>
+                <ul>
+                    <li>
+                        <a href="{{'/'}}">Home</a>
```
Comment (ghost) [inline resources/views/footer.blade.php:21]: you are over complicated here `"/"` is enough, but more correct is to have named route here

File: resources/views/pages/category.blade.php [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+                    </div>
+                @endforeach
+            </div>
+            <ul class="pagination">
+                {{$categories->links('paginator')}}
+            </ul>
+        </div>
+    </section>
+
+{{--        {{$categories}}--}}
```
Comment (ghost) [inline resources/views/pages/category.blade.php:56]: remove comment and all empty lines

**Review by ghost — COMMENTED** [review]
You did a great job, please apply all my comments, I don't write to all places but you should find out yourself and apply.

Docker-compose  is good to put in a gitignore and push to git a docker-compose.yml.example file.

db volume should be in gitignore

Also add in readme file all steps that should do for run application on local.

File: .idea/laravel-intro.iml [hunk 12]
```diff
@@ -116,6 +116,9 @@
       <excludeFolder url="file://$MODULE_DIR$/vendor/doctrine/event-manager" />
       <excludeFolder url="file://$MODULE_DIR$/vendor/doctrine/dbal" />
       <excludeFolder url="file://$MODULE_DIR$/vendor/doctrine/cache" />
+      <excludeFolder url="file://$MODULE_DIR$/.idea/dataSources" />
```
Comment (egorbwork) [inline .idea/laravel-intro.iml:119]: .idea folder should be stored in gitignore

File: app/Http/Middleware/RequestLoggingMiddleware.php [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+     * Handle an incoming request.
+     *
+     * @param Request $request
+     * @param Closure $next
+     * @return mixed
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        $this->requestLogger->logRequest($request);
+        if(app()->environment('local')){
```
Comment (egorbwork) [inline app/Http/Middleware/RequestLoggingMiddleware.php:29]: You should have separated logger and decide which one is needed by this check.

File: routes/web.php [hunk 14]
```diff
@@ -13,6 +19,12 @@
...
-});
+Route::get('/', [HomeController::class, 'index']);
+Route::get('article/{id}', [ArticleController::class, 'index']);
+Route::get('/category/{id}', [CategoryController::class, 'index']);
+Route::get('/tag/{id}', [TagController::class, 'index']);
+Route::get('/email', [ContactFormController::class, 'index']);
+Route::post('/email', [ContactFormController::class, 'sendMailas']);
+Route::get('/email', ['uses' => function () {
+    return view('mail.email');
+},'middleware'=> RequestLoggingMiddleware::class]);
```
Comment (egorbwork) [inline routes/web.php:30]: Middleware should be applied for all routes.

**Pull request merged**